Sources/NIO/BSDSocketAPI.swift

```diff
@@ -284,3 +461,533 @@ extension NIOBSDSocket.Option {
+    // from poll with reporting of failed connections (Connect Report 309411),
+    // which recommended that you use NetAPI instead.
+    extension NIOBSDSocket {
+        @inline(never)
+        public static func poll(fds: UnsafeMutablePointer<pollfd>, nfds: nfds_t,
+                                timeout: CInt) throws -> CInt {
+            return try syscall(blocking: false) {
+                cPoll(fds, nfds, timeout)
+            }.result
+        }
+    }
+#endif
+
+// TCP/IP Operations
+extension NIOBSDSocket {
+    @discardableResult
+    @inline(never)
+    internal static func inet_ntop(af Family: NIOBSDSocket.AddressFamily,
+                                   src pAddr: UnsafeRawPointer,
+                                   dst pStringBuf: UnsafeMutablePointer<CChar>,
+                                   size StringBufSize: socklen_t)
+            throws -> UnsafePointer<CChar> {
+        #if os(Windows)
+            // TODO(compnerd) use `InetNtopW` to ensure that we handle unicode properly
+            guard let result = WinSDK.inet_ntop(Family.rawValue, pAddr, pStringBuf,
+                                                Int(StringBufSize)) else {
+                throw IOError(windows: GetLastError(), reason: "inet_ntop")
+            }
+            return result
+        #else
+            return try wrapErrorIsNullReturnCall {
+                cInet_ntop(Family.rawValue, pAddr, pStringBuf, StringBufSize)
+            }
+        #endif
+    }
+
+    @inline(never)
+    internal static func inet_pton(af Family: NIOBSDSocket.AddressFamily,
+                                   src pszAddrString: UnsafePointer<CChar>,
+                                   dst pAddrBuf: UnsafeMutableRawPointer)
+            throws {
+        #if os(Windows)
+            // TODO(compnerd) use `InetPtonW` to ensure that we handle unicode properly
+            switch WinSDK.inet_pton(Family.rawValue, pszAddrString, pAddrBuf) {
+            case 0: throw IOError(errnoCode: EINVAL, reason: "inet_pton")
+            case 1: return
+            default: throw IOError(winsock: WSAGetLastError(), reason: "inet_pton")
+            }
+        #else
+            switch cInet_pton(Family.rawValue, pszAddrString, pAddrBuf) {
+            case 0: throw IOError(errnoCode: EINVAL, reason: "inet_pton")
+            case 1: return
+            default: throw IOError(errnoCode: errno, reason: "inet_pton")
+            }
+        #endif
+    }
+
+    @inline(never)
+    internal static func sendfile(socket s: NIOBSDSocket.Handle, fd: CInt,
+                                  offset: off_t,
+                                  len nNumberOfBytesToWrite: off_t)
+            throws -> IOResult<Int> {
+        #if os(iOS) || os(macOS) || os(tvOS) || os(watchOS)
+            var written: off_t = 0
+            do {
+                _ = try syscall(blocking: false) { () -> ssize_t in
+                    var w: off_t = off_t(nNumberOfBytesToWrite)
```

Comment:
Can we avoid the one-letter variable name here?

Reply:
This is just moved code.  I don't really care - if you want me to change the name, what would you like it changed to?

Reply:
Gross, let's fix it. The name of the parameter in the manual page is `len`, that would be appropriate.

Reply:
Hey, that's exactly what I did for the BSD socket APIs - the man page gives the names.